highlight.go

```diff
@@ -1,25 +1,95 @@
 package main
 
-import "regexp"
-
-var highlightPatternsRe = compileRegexps([]string{
-	`(fopen|hex2bin|die|exec|chr|hexdec|eval)\(`,
-	`\$_[A-Z]`,
-	`GLOBALS`,
-	`\S"\s*\.\s*"\S`, // " . "
-	`\S'\s*\.\s*'\S`, // ' . '
-	`base64_`,
-	`[a-zA-Z0-9\/\+\=]{40}`, // long base64? string
-	// `@(unlink|include|mysql)`, already more generic one below
-	// `../../..`, // too many fps
-	// `curl_exec,
-	`file_put_contents`,
-	`file_get_contents`,
-	`@[a-z_]{1,16}\(`, // suppressed function call
-	`\$.\(\$.\(`,
-	`call_user_func_array`,
-	`\@\$\w{1,12}\(`, // suppressed dynamic function call
-})
+import (
+	"bytes"
+	"regexp"
+)
+
+var (
+	highlightPatternsRe = compileRegexps([]string{
+		// php
+		`\$_[A-Z]`,       // $_GET, $_POST, etc.
+		`\S"\s*\.\s*"\S`, // " . "
+		`\S'\s*\.\s*'\S`, // ' . '
+		`@$?\w{1,16}\(`,  // suppressed function call
+		`\$.\(\$.\(`,     // $x($y(
+		`\@\$\w{1,12}\(`, // suppressed dynamic function call
+		`\/\*\s*\w+\s*\*\/.+\/\*\s*\w+\s*\*\/[^\s]+`,                       // comment obfuscation
+		`include\s{1,10}["'\x60](\w|\/)+\.(png|jpeg|svg|jpg|webp)["'\x60]`, // include php as image
+
+		// js
+		`(\\x[A-Z0-9]{2}){10,}`,
+
+		// common
+		`[a-zA-Z0-9\/\+\=]{25,}`, // long base64 string
+	})
+	highlightPatternsLit = [][]byte{
+		// php
+		[]byte(`system(`),
+		[]byte(`fopen(`),
+		[]byte(`hex2bin(`),
+		[]byte(`die(`),
+		[]byte(`chr(`),
+		[]byte(`hexdec(`),
+
+		[]byte(`exec`),
+		[]byte(`shell_exec`),
+		[]byte(`passthru`),
+		[]byte(`popen`),
+		[]byte(`proc_open`),
+		[]byte(`pcntl_exec`),
+		[]byte(`escapeshellcmd`),
+		[]byte(`preg_replace`),
+		[]byte(`create_function`),
+		[]byte(`call_user_func_array`),
+
+		[]byte(`base64_`),
+		[]byte(`str_rot13`),
+		[]byte(`htmlspecialchars_decode`),
+
+		[]byte(`file_get_contents`),
+		[]byte(`file_put_contents`),
+		[]byte(`fwrite`),
+		[]byte(`fread`),
+		[]byte(`fgetc`),
+		[]byte(`fgets`),
+		[]byte(`fscanf`),
+		[]byte(`fgetss`),
+		[]byte(`fpassthru`),
+		[]byte(`readfile`),
+
+		[]byte(`gzuncompress`),
+		[]byte(`gzinflate`),
+		[]byte(`gzdecode`),
+		[]byte(`readgzfile`),
+		[]byte(`gzwrite`),
+		[]byte(`gzfile`),
+
+		[]byte(`umask(0)`),
+		[]byte(`chmod($`),
+		[]byte(`chown($`),
+		[]byte(`chgrp($`),
+		[]byte(`unlink(`),
+		[]byte(`rmdir(`),
+		[]byte(`mkdir(`),
+		[]byte(`stream_get_meta_data`),
+
+		[]byte(`GLOBALS`),
+
+		[]byte(`$obirninja`),
+		[]byte(`$pass`),
```

Comment:
```suggestion
```

Possibly too many FPs (it's supposed to match password variables in reverse shells, etc... - should probably make it more specific then)

Reply:
IMO better to highlight a FP than to not highlight a TP.

Let's be more greedy :)

Reply:
IMO better to highlight a FP than to not highlight a TP.

Let's be more greedy :)